environs/config/config.go

```diff
@@ -100,6 +108,23 @@ const (
 	// NoProxyKey stores the key for this setting.
 	NoProxyKey = "no-proxy"
 
+	// The new proxy keys are passed into hook contexts with the prefix
+	// JUJU_CHARM_ then HTTP_PROXY, HTTPS_PROXY, FTP_PROXY, and NO_PROXY.
+	// This allows the charm to set a proxy when it thinks it needs one.
+	// These values are not set in the general environment.
+
+	// JujuHTTPProxyKey stores the key for this setting.
+	JujuHTTPProxyKey = "juju-http-proxy"
+
+	// JujuHTTPSProxyKey stores the key for this setting.
+	JujuHTTPSProxyKey = "juju-https-proxy"
+
+	// JujuFTPProxyKey stores the key for this setting.
+	JujuFTPProxyKey = "juju-ftp-proxy"
+
+	// JujuNoProxyKey stores the key for this setting.
+	JujuNoProxyKey = "juju-no-proxy"
+
 	// AptHTTPProxyKey stores the key for this setting.
 	AptHTTPProxyKey = "apt-http-proxy"
```

Comment:
To aid the reader, worth a comment why apt proxy is not legacy but is not given the JUJU_ prefix?